README.md

```diff
@@ -133,6 +137,75 @@ Assets and events will ignore existing ids on deletes. If you prefer to abort th
 when an attempt to delete an unknown id, use `.option("ignoreUnknownIds", "true")`
 for those resources types.
 
+## Asset hierarchy (beta)
+Note: The asset hierarchy utility is currently in beta, and has not been sufficiently tested to be used 
+on production data.
+
+The `.option("type", "assethierarchy")` lets you write new asset hierarchies, or update existing ones,
+using the Spark data source. It currently supports writing a single root asset and any number of children.
+The asset hierarchy utility can ingest entire trees, as long as all nodes are connected to the root asset
+through the `externalId`/`parentExternalId` relationship. If input contains an update to data that already exists,
+i.e there's a match on `externalId` and there's a change to one of the other fields, the asset will be updated.
+There's also an option to delete assets from CDF that are not referenced in the input data.
+
+### Requirements
+- The source data must contain a single root, denoted by setting its `parentExternalId` to the empty string `""`.
+- All assets, except root, must be connected to another asset in the input data by `parentExternalId`.
+- The input data must not have loops, to ensure all asset trees are fully connected.
+- `externalId` can not be the empty string `""`.
+
+### Options
+|    Option                  | Default | Description |
+| -------------------------- | --------|----------------------------------------------------------------------------------------------------------- |
+| `deleteMissingAssets`      | `false` | Whether or not you would like assets under the root to be deleted if they're not present in the input data |
+| `ignoreDisconnectedAssets` | `false` | This will ignore assets that are not connected to root, as opposed to throw an error and stop the program. |
+| `batchSize`                | 1000    | The number of assets to write per API call                                                                 |
+
+### Example
+```scala
+val assetHierarchySchema = Seq("externalId", "parentExternalId", "description", "name", "metadata")
+
+// Manually create some assets that satisfy the requirements of the asset hierarchy utility
+val assetTree = Seq(
+  Seq("root_asset", "", "manual_input", "root_asset", Some("This is the root asset"), None),
+  Seq("first_child", "root_asset", "manual_input", "first_child", Some("This is the first_child"), None),
+  Seq("second_child", "root_asset", "manual_input", "second_child", Some("This is the second_child"), None),
+  Seq("grandchild", "", "manual_input", "first_child", Some("This is the root asset"), None)
+)
+
+val assetTreeDataFrame = spark
+  .sparkContext
+  .parallelize(assetTree)
+  .toDF(assetHierarchySchema) // The column names
+
+// Insert the assets with the asset hierarchy tool
```

Comment:
I see we're using the term "asset hierarchy utility" elsewhere in this article, so let's be consistent about what we call it :)

Reply:
I don't really like utility either, I just chose it because I thought builder _could_ be misleading. In retrospect I like builder more anyways, so let's go with that.